### ISSUE TITLE
Short thread IDs

### ISSUE THREAD

**Issue by Timmmm** [gabime/spdlog#455]
On OSX, thread IDs are *long*, like `3817087084462213816`. I wonder if there is a way to shorten them. Hex is slightly shorter, but what I really want is just a low integer for each thread. Something like this but fast and thread safe (good luck with that!):

```
int next_thread_short_id = 1;
std::map<size_t, int> thread_short_ids;

int getThreadShortId(size_t thread_id)
{
    if (thread_short_ids.count(thread_id) == 0)
        thread_short_ids[thread_id] = next_thread_short_id++;
    return thread_short_ids[thread_id];
}
```

**Comment by gabime** [comment]
Fixed (partially?) in 431fb423fa1323465a41a63312f91aca5f0328ea

**Comment by Timmmm** [comment]
Much better. It would really by nice to have ones like `1`, `2, `3` but I'll close this anyway.
